src/main/java/software/amazon/cloudformation/proxy/AmazonWebServicesClientProxy.java

```diff
@@ -174,23 +250,54 @@ public ClientT client() {
                         }
 
                         @Override
-                        public Completed<RequestT, ResponseT, ClientT, ModelT, CallbackT>
-                            exceptFilter(Callback<? super RequestT, Exception, ClientT, ModelT, CallbackT, Boolean> handler) {
-                            return exceptHandler((request, exception, client1, model1, context1) -> {
-                                if (handler.invoke(request, exception, client1, model1, context1)) {
-                                    return ProgressEvent.progress(model1, context1);
+                        public Completed<RequestT, ResponseT, ClientT, ModelT, CallbackT> retryErrorFilter(final Callback<
+                            ? super RequestT, Exception, ClientT, ModelT, CallbackT, Boolean> retryFilter) {
+                            return handleError(((request, exception, client_, model_, context_) -> {
+                                if (retryFilter.invoke(request, exception, client_, model_, context_)) {
+                                    throw RetryableException.builder().build();
```

Comment:
so this exception will trigger a retry? is it based on number of attempts (if so that might be different from ln:181 as that returns `progress`) or the delay?

Reply:
The model is as follows. handleError can either re-throw the exception or return a ProgressEvent. If we want to retry it throws a RetryableException which attempts it once more. This code is delegating to the more general handleError function using a simple predicate to say yes or no to where the exception can be retried